src/porepy/models/constitutive_laws.py

```diff
@@ -2462,22 +2495,80 @@ def matrix_porosity(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
             Cell-wise porosity operator [-].
 
         """
+
+        # Sanity check
         if not all([sd.dim == self.nd for sd in subdomains]):
             raise ValueError("Subdomains must be of dimension nd.")
+
+        # Add contributions to poromechanics porosity
+        phi = (
+            self.reference_porosity(subdomains)
+            + self.pressure_contribution(subdomains)
+            + self.displacement_divergence_contribution(subdomains)
+            + self.biot_stabilization(subdomains)
+        )
+        phi.set_name("Stabilized matrix porosity")
+
+        return phi
+
+    def reference_porosity(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Reference porosity.
+
+        Parameters:
+            subdomains: List of subdomains where the reference porosity is defined.
+
+        Returns:
+            Reference porosity operator.
+
+        """
+        return Scalar(self.solid.porosity(), "reference_porosity")
+
+    def pressure_contribution(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Contribution of the pressure changes to the matrix porosity [-].
+
+        Parameters:
+            subdomains: List of subdomains where the porosity is defined.
+
+        Returns:
+            Cell-wise contribution of the pressure changes to the matrix porosity [-].
+
+        """
+
+        # Retrieve material parameters
+        alpha = self.biot_coefficient(subdomains)
         phi_ref = self.reference_porosity(subdomains)
+        bulk_modulus = self.bulk_modulus(subdomains)
+
+        # Pressure changes
         dp = self.perturbation_from_reference("pressure", subdomains)
-        alpha = self.biot_coefficient(subdomains)
-        bulk = self.bulk_modulus(subdomains)
 
-        # 1/N as defined in Coussy, 2004, https://doi.org/10.1002/0470092718.
-        n_inv = (alpha - phi_ref) * (1 - alpha) / bulk
+        # Compute 1/N as defined in Coussy, 2004, https://doi.org/10.1002/0470092718.
+        n_inv = (alpha - phi_ref) * (1 - alpha) / bulk_modulus
 
-        # Add the three contributions to the porosity.
-        phi = phi_ref + n_inv * dp + alpha * self.displacement_divergence(subdomains)
-        # Add stabilization term. TODO: Should this be handled elsewhere?
-        phi = phi + self.biot_stabilization(subdomains)
-        phi.set_name("stabilized_matrix_porosity")
-        return phi
+        # Pressure change contribution
+        pressure_contribution = n_inv * dp
+        pressure_contribution.set_name("Pressure contribution to porosity")
```

Comment:
We should agree on format for operator names just for the sake of uniformity. Let's discuss briefly.

Reply:
Done.